### DISCUSSION TITLE
How to use Framework generated token in client sdk?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#3114] [Q&A]
In framework, using token package of framework we can generate token for user in custom endpoints by validating it with thirdparty api's

```go
token, err := tokens.NewRecordAuthToken(app, authUserRecord)
if err != nil {
	return err
}

```

But how to use the generated token in Flutter SDK/JS SDK and authenticate the user?

**Comment by ganigeorgiev** [comment]
I'm not sure if that is what you are asking but you can load a token into the SDKs AuthStore by calling `pb.authStore.save(token, model?)`.

Additionally, if you have a custom auth route, you can also return directly an auth response with [`apis.RecordAuthResponse(app, c, authUserRecord, nil)`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.17.3/apis#RecordAuthResponse) (_it will generate the token for you and the returned response will be in the same format as when authenticating with the builtin actions + will trigger the `OnRecordAuthRequest` hook_).
